geoutils/raster/raster.py

```diff
@@ -3363,96 +3443,186 @@ def split_bands(self: RasterType, copy: bool = False, bands: list[int] | int | N
         self,
+        data_column_name: str = "b1",
+        data_band: int = 1,
+        store_auxiliary_bands: bool = False,
         subsample: float | int = 1,
+        *,
         as_array: bool = False,
-        pixel_offset: Literal["center", "corner"] = "center",
+        random_state: np.random.RandomState | int | None = None,
+        force_pixel_offset: Literal["center", "ul", "ur", "ll", "lr"] = "ul",
+    ) -> NDArrayNum | Vector:
+        ...
+
+    def to_pointcloud(
+        self,
+        data_column_name: str = "b1",
+        data_band: int = 1,
+        store_auxiliary_bands: bool = False,
+        subsample: float | int = 1,
+        as_array: bool = False,
+        random_state: np.random.RandomState | int | None = None,
+        force_pixel_offset: Literal["center", "ul", "ur", "ll", "lr"] = "ul",
     ) -> NDArrayNum | Vector:
         """
-        Convert raster to a table of coordinates and their corresponding values.
+        Convert raster to point cloud.
+
+        A point cloud is a vector of point geometries associated to a data column, and possibly other auxiliary data
+        columns, see geoutils.PointCloud.
 
-        Optionally, randomly sample the raster.
+        For a single band raster, the main data column name of the point cloud defaults to "b1" and stores values of
+        that single band.
+        For a multi-band raster, the main data column name of the point cloud defaults to "bX" where X is the data band
+        index chosen by the user (defaults to 1, the first band).
+        Optionally, all other bands can also be stored in columns "b1", "b2", etc. For more specific band selection,
+        use Raster.split_bands previous to converting to point cloud.
 
-        If 'sample' is either 1, or is equal to the pixel count, all points are returned in order.
-        If 'sample' is smaller than 1 (for fractions), or smaller than the pixel count, a random sample
-        of points is returned.
+        Optionally, randomly subsample valid pixels for the data band (nodata values are skipped, but only for the band
+        that will be used as data column of the point cloud).
+        If 'subsample' is either 1, or is equal to the pixel count, all valid points are returned.
+        If 'subsample' is smaller than 1 (for fractions), or smaller than the pixel count, a random subsample
+        of valid points is returned.
 
-        If the raster is not loaded, sampling will be done from disk without loading the entire Raster.
+        If the raster is not loaded, sampling will be done from disk using rasterio.sample after loading only the masks
+        of the dataset.
 
         Formats:
             * `as_array` == False: A vector with dataframe columns ["b1", "b2", ..., "geometry"],
             * `as_array` == True: A numpy ndarray of shape (N, 2 + count) with the columns [x, y, b1, b2..].
 
+        :param data_column_name: Name of point cloud data column to use, defaults to "b1".
```

Comment:
I would rephrase: "Name to use for point cloud data column" or it sounds like we need to specify the name from some inputs.

Reply:
Done